src/encryption/api.go

```diff
@@ -0,0 +1,75 @@
+package encryption
+
+import (
+	"context"
+	"fmt"
+	"keboola-as-code/src/client"
+	"strings"
+
+	"github.com/go-resty/resty/v2"
+	"go.uber.org/zap"
+)
+
+type EncryptionApi struct {
+	apiHost string
+	client  *client.Client
+	logger  *zap.SugaredLogger
+}
+
+// Error represents Encryption API error structure
+type Error struct {
+	Message     string `json:"error"`
+	ErrCode     int    `json:"code"`
+	ExceptionId string `json:"exceptionId"`
+}
+
+func (e *Error) Error() string {
+	msg := fmt.Sprintf(`"%v", errCode: "%v"`, e.Message, e.ErrCode)
+	if len(e.ExceptionId) > 0 {
+		msg += fmt.Sprintf(`, exceptionId: "%s"`, e.ExceptionId)
+	}
+	return msg
+}
+
+func getEncryptionApiHost(connectionApiHost string) string {
+	return strings.ReplaceAll(connectionApiHost, "connection.", "encryption.")
```

Comment:
potrebujem zistit url na encryption api, to urobim tak ze v url pre storage api replacnem "connection" za "encryption"

Reply:
toto si myslim nie je OK, ... mozno to aktualne plati pre vsetky stacks ale nespoliehal by som sa.

Myslim, ze by sa mal pouzit API call na Storage API:
```
curl --include \
'https://connection.keboola.com/v2/storage?exclude=components'
```

Vysledok je:
```
{
    "...": "...",
    "services": [
        {
            "id": "encryption",
            "url": "https://encryption.keboola.com"
        }
    ],
    "....": "...."
}
```

Teda do storage API pridat call na ziskanie encryption API host.
A ten host zadat ako parameter pri vytvarani `encryption.Api`.

Reply:
jj suhlas, nacitam to cez storage api services